internal/stacks/stack.go

```diff
@@ -533,19 +532,18 @@ func (s *Stack) deployContracts(spin *spinner.Spinner, verbose bool) error {
 
 			contractDeployed = true
 		} else {
-			// Just load the ABI
 			updateStatus(fmt.Sprintf("publishing FireFly ABI to '%s'", member.ID), spin)
 			publishFireflyResponse, err := contracts.PublishABI(ethconnectUrl, fireflyContract)
 			if err != nil {
 				return err
 			}
-			fireflyAbiId = publishFireflyResponse.ID
-		}
-		// Register as "firefly"
-		updateStatus(fmt.Sprintf("registering FireFly contract on '%s'", member.ID), spin)
-		_, err := contracts.RegisterContract(ethconnectUrl, fireflyAbiId, fireflyContractAddress, member.Address, "firefly", map[string]string{"paymentContract": paymentContractAddress})
-		if err != nil {
-			return err
+			fireflyAbiId := publishFireflyResponse.ID
+
```

Comment:
Was moving the registration inside the `else` intentional? I'm pretty sure it needs to be registered on each `ethconnect` which is why it was outside the if/else before.

Reply:
Yes, fully intentional. It's in the commit message (but I know GitHub doesn't really surface the full text well). Basically if you call "deploy" and pass a registered name, it does a deploy+register in one step, and calling registration _again_ on that node generates a 409 (and fails my new checks). So we only need to register on the non-deployer nodes.

Reply:
Understood! Makes sense. Thanks!